src/GraphQL/Language/AST/Field.cs

```diff
@@ -88,29 +88,17 @@ public override IEnumerable<INode> Children
         /// <summary>
         /// Determines if this instance is equal to another instance by comparing the <see cref="Name"/> and <see cref="Alias"/> properties.
         /// </summary>
-        protected bool Equals(Field other)
+        private bool Equals(Field other)
         {
             return string.Equals(Name, other.Name, StringComparison.InvariantCulture) && string.Equals(Alias, other.Alias, StringComparison.InvariantCulture);
         }
 
-        /// <inheritdoc/>
-        public override bool IsEqualTo(INode obj)
-        {
-            if (obj is null)
-                return false;
-            if (ReferenceEquals(this, obj))
-                return true;
-            if (obj.GetType() != GetType())
-                return false;
-            return Equals((Field)obj);
-        }
-
         /// <summary>
         /// Returns a new field selection node with the child field selection nodes merged with another field's child field selection nodes.
         /// </summary>
         public Field MergeSelectionSet(Field other)
         {
-            if (Equals(other))
+            if (Equals(other)) //TODO: is this check necessary?
```

Comment:
I suggest to remove this method and unwrap its body inside single consumer: `Fields.Add(Field field)`

Reply:
Also remove Equals of course

Reply:
I agree

Reply:
Done